joomla/joomla-cms
# [4.1] Fix the upgrade  path

Pull Request for Issue # .

### Summary of Changes
- Introduced a function in the `script.php` that will run independently of the batch work of the db updates. The reason is that this particular db fix is essential so that the UI will not become broken if for any reason the update files fail somehow

### Testing Instructions
Using https://test5.richard-fath.de/Joomla_4.1.0-dev-Development-Update_Package_test-with-sql-error.zip try to
- Update from 3.10.x to the 4.1  => actual messed result
- Update from 4.0.x to the 4.1  => actual messed result
(these **SHOULD FAIL** leaving you with unstyled front/backend templates)

Using https://test5.richard-fath.de/Joomla_4.1.0-dev+pr.36585-Development-Update_Package_test-with-sql-error.zip try to
- Update from 3.10.x to the 4.1  => expected usable result
- Update from 4.0.x to the 4.1 => expected usable result
(these should show an SQL error at the end, but the backend is usable and the db fixer can be used)


### Actual result BEFORE applying this Pull Request

When the database update fails before the update for child templates would have run, the UI is unusable.

![j4 1-test-pr-36585_after-update-from-4 0-without-patch](https://user-images.githubusercontent.com/7413183/150871146-5a183c64-98d0-4b10-9d08-efd2d3655844.png)

### Expected result AFTER applying this Pull Request

The UI is usable. An SQL error is shown, but the backend works, and the database fixer can be used.

![j4 1-test-pr-36585_after-update-from-4 0-with-patch](https://user-images.githubusercontent.com/7413183/150871396-90ebaa88-8dcb-4c7e-99a6-0e364546f6c9.png)

### Documentation Changes Required

@brianteeman @joeforjoomla could you test this one?

@richard67 I hope I didn't messed up here
